File: app/src/main/java/com/botijasoftware/utils/interpolator/SmoothStepInterpolator.java
package com.botijasoftware.utils.interpolator;

public class SmoothStepInterpolator implements Interpolator {
	//public abstract Interpolator() {}
	public float interpolate(float v0, float v1, float t) {
		float a = (3 - 2*t)*t*t;
		return v0 + a*(v1 - v0);
	}
}

File: app/src/main/java/com/botijasoftware/utils/interpolator/SmoothStepInterpolator.kt
package com.botijasoftware.utils.interpolator

class SmoothStepInterpolator : Interpolator {
    //public abstract Interpolator() {}
    override fun interpolate(v0: Float, v1: Float, t: Float): Float {
        val a = (3 - 2 * t) * t * t
        return v0 + a * (v1 - v0)
    }
}
